### PR TITLE
Fix reported issues w.r.t python2.7 and some apple silicone quirks

### PR DESCRIPTION
**Pull request by jtrmal**
@danpovey  for your reviewing pleasure ;)

### CODE REVIEW

File: src/util/parse-options-test.cc [hunk 1]
```diff
@@ -120,7 +120,7 @@ void UnitTestParseOptions() {
     po4.Register("option", &val, "My boolean");
     po4.Read(argc4, argv4);
     assert(false); // Should not reach this part of code.
-  } catch(std::exception e) {
+  } catch(std::exception &e) {
```
Comment (csukuangfj) [inline src/util/parse-options-test.cc:123]: I suggest using `const std::exception &e`.

`const` makes the intention clear that it is not intended to be modified.

**Comment by jtrmal** [comment]
@csukuangfj  hi, good point, changed. I also fixed the failing tests while I was at it -- authors of the tests forgot that `KALDI_ASSERT(cond)` calls/evaluates `cond` twice. LMK

**Comment by danpovey** [comment]
I'm surprised some of these issues lasted as long as they have, about the templates.  anyway looks OK to me.

**Comment by jtrmal** [comment]
I'm surprised too, but some stuff we ignored and some stuff the older compilers didn't complain about... 🤷

**Comment by boeddeker** [comment]
Is there a reason, that `mkdir -p $PWD/python` was removed in `check_dependencies`?
My kaldi install script is currently failing, and I get `ln: failed to create symbolic link '.../kaldi/tools/python/python'*` warnings. I will fix it in my script, but I assume it would be good to have check_dependencies to be robust.

**Comment by jtrmal** [comment]
Might had been unintentional change, are you willing to create a PR?
Thanks!
y.

**Pull request merged**